parallel-install/pkg/deployment/overrideinterceptors.go

```diff
@@ -121,6 +168,75 @@ func (i *DomainNameOverrideInterceptor) findLocalDomain() (domainName string, er
 	return "", nil
 }
 
+// patchCoreDNS takes kubeclient and cluster domain as a parameter to patch coredns config
+// e.g. domainName: `(.*)\.local\.kyma\.dev`
+func patchCoreDNS(kubeClient kubernetes.Interface, domainName string, log logger.Interface) (err error) {
+	coreDNSPatch := CoreDNSPatch{DomainName: domainName}
+	patchTemplate := template.Must(template.New("").Parse(coreDNSPatchTemplate))
+	patchBuffer := new(bytes.Buffer)
+	if err = patchTemplate.Execute(patchBuffer, coreDNSPatch); err != nil {
+		return err
+	}
+
+	// TODO: Refactor
+	err = retry.Do(func() error {
+		_, err := kubeClient.AppsV1().Deployments("kube-system").Get(context.TODO(), "coredns", metav1.GetOptions{})
```

Comment:
Extract to a separate function that is responsible for coreDNS detection.

Reply:
Done